xds/src/main/java/com/linecorp/armeria/xds/ConfigSourceMapper.java

```diff
@@ -91,22 +84,28 @@ ConfigSource cdsConfigSource(ResourceNode<?> node, String resourceName) {
         throw new IllegalArgumentException("Cannot find a CDS config source for route: " + resourceName);
     }
 
-    ConfigSource rdsConfigSource(ResourceNode<?> node, String resourceName) {
-        final ConfigSource configSource = node.configSource();
+    ConfigSource rdsConfigSource(@Nullable ConfigSource parentConfigSource, @Nullable ConfigSource configSource,
+                                 String resourceName) {
         if (configSource != null && configSource.hasApiConfigSource()) {
             return configSource;
         }
+        if (configSource != null && configSource.hasSelf() && parentConfigSource != null) {
+            return parentConfigSource;
+        }
         if (bootstrapAdsConfig != null) {
             return bootstrapAdsConfig;
         }
         throw new IllegalArgumentException("Cannot find an RDS config source for route: " + resourceName);
     }
 
-    ConfigSource ldsConfigSource(ResourceNode<?> node) {
-        final ConfigSource configSource = node.configSource();
+    ConfigSource ldsConfigSource(@Nullable ConfigSource parentConfigSource,
+                                 @Nullable ConfigSource configSource) {
```

Comment:
Should we also accept the `resourceName` to include it in the exception as other methods do?